deps/wazuh_testing/wazuh_testing/tools/performance/statistic.py

```diff
@@ -101,53 +124,478 @@ def _parse_logcollector_state_file(self, data):
             else:
                 self._parse_logcollector_state_file(data)
         except Exception as e:
             logger.error(f'Exception with {self.statistics_file} | {str(e)}')
 
-    @staticmethod
-    def _write_csv(data, csv_file):
+
+    def _parse_api_data(self):
+        """Read the data from the statistics file generated by Wazuh API."""
+
+        API_URL = f"https://{self.ip}:{self.port}"
+        DAEMONS_ENDPOINT= f"/manager/daemons/stats?daemons_list={self.daemon}"
+        TOKEN_ENDPOINT="/security/user/authenticate"
+
+        logging.info("Getting statistics data from API for {}".format(self.target))
+
+        max_retries = 3
+        token_response = None
+        daemon_response = None
+        # Try to get the response token three times
+        for _ in range(max_retries):
+            try:
+                token_response = requests.get(API_URL + TOKEN_ENDPOINT, verify=False,
+                                              auth=requests.auth.HTTPBasicAuth("wazuh", "wazuh"))
+                if token_response.status_code != 200:
+                    break
+            except requests.exceptions.RequestException as e:
+                logging.error(f"Error getting token from API: {str(e)}")
+        else:
+            logging.error("Retrying get API data, status code {}".format(token_response.status_code))
+
+        for _ in range(max_retries):
+            try:
+                daemons_response = requests.get(API_URL + DAEMONS_ENDPOINT, verify=False,
+                                                headers={'Authorization': 'Bearer ' + token_response.json()['data']['token']})
+                if daemons_response.status_code != 200:
+                    break
+            except requests.exceptions.RequestException as e:
+                logging.error(f"Error fetching {self.daemon} datafrom API: {str(e)}")
+        else:
+             logging.error("Failed to fetch daemons data after 3 attempts")
+        
+
+        data = daemons_response.json()['data']['affected_items'][0]
+        self._write_csv(data, self.target, self.csv_file)
+
+
+    def _write_csv(self, data, target, csv_file):
         """Write the data collected from the .state into a CSV file.
 
         Args:
             data (dict): dictionary containing the info from the .state file.
+            target (string): specifies which CSV must be generated
             csv_file (string): path to the CSV file.
         """
+        if target == "analysis":
+            csv_header = headers.analysisd_header if self.use_state_file == True else headers.analysisd_events_header
+        elif target == "logcollector":
+            csv_header = headers.logcollector_header
+        elif target == "remote":
+            csv_header = headers.remoted_header if self.use_state_file == True  else headers.remoted_api_header
+        elif target == "wazuhdb":
+            csv_header = headers.wazuhdb_header
+        else:
+            csv_header = headers.agentd_header
```

Comment:
```suggestion
        if target == "analysis":
            csv_header = headers.analysisd_header if self.use_state_file else headers.analysisd_events_header
        elif target == "logcollector":
            csv_header = headers.logcollector_header
        elif target == "remote":
            csv_header = headers.remoted_header if self.use_state_file else headers.remoted_api_header
        elif target == "wazuhdb":
            csv_header = headers.wazuhdb_header
        else:
            csv_header = headers.agentd_header
```

Reply:
Fixed in https://github.com/wazuh/wazuh-qa/pull/4970/commits/b48cd5b48740e5e2721a0809757ed50ae0411f79